cognite/neat/rules/transformers/_mapping.py

```diff
@@ -95,3 +96,33 @@ def transform(self, rules: DMSRules | OutRules[DMSRules]) -> JustRules[DMSRules]
                 prop.reference = ReferenceEntity.from_entity(ref_prop.view, ref_prop.view_property)
 
         return JustRules(solution)
+
+
+class RuleMapper(RulesTransformer[InformationRules, InformationRules]):
+    """Maps properties and classes using the given mapping.
+
+    **Note**: This transformer mutates the input rules.
+
+    Args:
+        mapping: The mapping to use.
+
+    """
+
+    def __init__(self, mapping: RuleMapping) -> None:
+        self.mapping = mapping
+
+    def transform(self, rules: InformationRules | OutRules[InformationRules]) -> JustRules[InformationRules]:
```

Comment:
Switched this to operate on verified rules instead of input rules. The motivation is that since mapping is a pydantic class and already verified to ClassEntity, the code becomes a lot cleaner. Mapping is also something you probably do after you have validated, still up for discussion.